src/routes/article.ts

```diff
@@ -0,0 +1,42 @@
+import clownface from 'clownface';
+import createHttpError from 'http-errors';
+import { OK } from 'http-status-codes';
+import { addAll } from 'rdf-dataset-ext';
+import {
+  DefaultStateExtends, Middleware, Next,
+} from 'koa';
+import url from 'url';
+import Articles from '../articles';
+import { namedNode } from '../data-factory';
+import ArticleNotFound from '../errors/article-not-found';
+import { DatasetContext } from '../middleware/dataset';
+
+export default (articles: Articles): Middleware<DefaultStateExtends, DatasetContext> => (
+  async ({ request, response }: DatasetContext, next: Next): Promise<void> => {
+    try {
+      await next();
+    } catch (error) {
+      if (error instanceof createHttpError.NotFound) {
```

Comment:
Should be able to return early from both a successful `next()` and an error that isn't a `NotFound`.

Reply:
Think might be able to pass this as a 'not found' handler to the routing middleware, so this middleware itself doesn't have to handle it.

Reply:
> Think might be able to pass this as a 'not found' handler to the routing middleware, so this middleware itself doesn't have to handle it.

Isn't the routing middleware run before the get article request is handled. How will the 'not found' handler be passed to the routing middleware unless we amend that workflow?